text/0000-compiler-team-trusted-contrib-maintainer.md

```diff
@@ -0,0 +1,381 @@
+
+Definitions
+-----------
+[definitions]: #definitions
+
+There are various permissions/privileges/responsibilities which will be
+referenced in later sections of this RFC, defined here:
+
+- **r+**
+  - Contributors with *r+* privilege are able to approve pull requests to be
+    merged by *bors*. Contributors should not merge their own pull requests (with
+    the exception of re-approving their own work on behalf of another contributor
+    after a rebase or similarly trivial change). *r+* permissions apply to the
+    whole repository, but [it is expected][expectations] that contributors limit
+    themselves to only those parts of the *rust-lang/rust* repository that are under
+    the purview of the compiler team (unless granted *r+* from other teams too), and
+    for subsystems/pull requests that they are confident reviewing.
+- **try**
+  - Contributors with *try* permissions are able to trigger complete toolchain
+    builds for a pull request or commit, which are then used by *rustc-perf* and
+    *crater*.
+- **review queue**
+  - Contributors on the review queue will be randomly assigned to new pull
+    requests submitted to the compiler. Being on the review queue is one of the
+    best ways for contributors to help the compiler team and learn new parts of
+    the compiler. Review capacity is one of the most important resources that the
+    team has, as it enables our progress in the compiler's continued development
+    and maintenance.
+- **organization membership**
+  - Contributors that are added to the *rust-lang/compiler* team in the GitHub
+    organisation can be assigned to issues/pull requests, modify labels, receive
+    group mentions and receive a "Member" badge next to their name.
+- **rustc-perf**
+  - Contributors with permissions to use *rustc-perf* can request benchmarking
+    of their pull requests (and pull requests they are reviewing). *rustc-perf*
+    permissions are useful for regular contributors as it is common to need to
+    request benchmarks from contributors with permissions. *rustc-perf* permissions
+    only make sense alongside *try* permissions.
+- **crater**
+  - Contributors with permissions to *crater* can request crater runs to check
+    whether their code breaks any public ecosystem code. *crater* permissions only
+    make sense alongside *try* permissions.
+- **dev desktops**
+  - Contributors with access to developer desktops are able to connect to shared
+    development servers that they can do their contributions from.
+
+[expectations]: https://forge.rust-lang.org/compiler/reviews.html#expectations-for-r
+
+Guide-level explanation
+=======================
+[guide-level-explanation]: #guide-level-explanation
+
+Contributors start without any particular privileges, permissions or
+responsibilities and can contribute whatever they'd like. Contributors can
+progress to [Trusted Contributors][trusted-contributors] and then [Team Members]
```

Comment:
I've very much warmed up to the idea of changing the naming to "contributor" -> "member" (from trusted contributor) -> "maintainer" (from member).

First, the "maintainer" name does seem to represent better what these members (participants? we somewhat overload this word in English) are tasked to do.

For "Member" vs contributor, see the language: What does someone interested who submits a couple PRs to the rust repo do? They *contribute*. If we give them permissions for bors and such, and do various things, they're now a *member* of whatever team that is.

Ultimately though, its just naming. And the name does differ quite a bit from the current naming. But I think it better matches what we expect when we use these words in everyday English.

Reply:
I agree, this aligns with my own thinking on the topic more too. My only concern is that it diverges from what other teams have a little bit. I've renamed these in the RFC.